include/bbp/sonata/config.h

```diff
@@ -208,11 +208,11 @@ class SONATA_API SimulationConfig
         /// Report type. Possible values: "compartment", "summation", "synapse"
         std::string type;
         /// Interval between reporting steps in milliseconds
-        float dt{};
+        double dt{};
```

Comment:
`dt`/`startTime`/`endTime` are all floats in the spec for the report, so I think it makes sense to keep the same type:

https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md#simulation-output---reports

thoughts @NadirRoGue?

Reply:
Unless precision is a problem, I prefer floats as well

Reply:
Yes, precision is a problem for neurodamus. While reading out the floating values from libsonata, such as `dt`, the precision is different from the original value in the json file, which leads to different simulation results. As python uses double precision, as well as the json parser https://json.nlohmann.me/api/basic_json/number_float_t/ . I think it is better to align.

Reply:
> which leads to different simulation results

How does it lead to different results?
From the spec for the reports, the values are in floats, so how is that conversion taking place safely?

Reply:
Hello @mgeplf (cc @pramodk @ferdonline ) , in python and also py-neurodamus when I read a simulation config file containing `"dt": 0.025` via libsonata (master branch), I get this value:
```
>>> from libsonata import SimulationConfig
>>> conf = SimulationConfig.from_file("simulation_config.json")
>>> conf.run.dt
0.02500000037252903
>>> print("%.15f" %conf.run.dt)
0.025000000372529
```
neurodamus passes it to NEURON which also uses double type for floating, and the result is different (tested by CI). I see that the SONATA spec uses `float` type for all time variables, and on the other side the parser `nlohmann::json` uses `double`.  Although in python I could correct that value by `numpy.float32(dt)`, I think it would be better that libsonata outputs the right value. What do you think?

BTW, maybe it would work by changing the default type to `float` in the parser, as mentioned [here](https://github.com/nlohmann/json/issues/1109#issuecomment-392255532). But it didn't work for me, still trying.

Reply:
```
>>> print("%.15f" %conf.run.dt)
0.025000000372529
```
comes from printing `%.15f`, if `%.17f' is used, you get the same as the first case: 0.02500000037252903

Doubles (float64) are a strict superset of float32, so converting from float32 -> to float64 should always give the same representation.  Once it's been in float32, it should always be the same:
Ex: https://godbolt.org/z/fzMGz17Wx

That means that if the float64 parsed from nlohmann gets cast to float32 in libsonata, and should be the same in all cases, whether it's read as a float32 or float64.

If the simulation you're comparing in to in the CI was using a different a different parser to get the dt, it might always have been in float64, and thus the number will be different.  That's why I wrote above that keeping the same type as the expected report output makes sense: otherwise that conversion will happen at some other point, and it won't be easy/possible to reconcile what was written to the report versus what was simulated.

Reply:
Actually the report saves this info as double as per spec:
https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md#output-file-formats

So it will be converted to double for the creation of the report anyway...

Reply:
float->double is not a problem, but reading as float is the step where we lose some decimal places and enter different results land.
Being very objective, using doubles will not make a difference to anyone except ourselves who are bound to ensure reproducibility. And this being the case I think it makes sense to change to double to ensure a smooth transition.
Even if someone really wants a float, well, truncating is always possible, just the opposite is not.

Reply:
@jorblancoa > Actually the report saves this info as double as per spec: 

Yup, that's the important one - I missed that:
`/report/<population_name>/mapping/time (dtype: double, shape: 3): the values of the data set are start time, stop time and time step`

`double` it is.